packages/docusaurus/src/client/RouteAnnouncer.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import {Portal} from './exports/Portal';
+
+type Props = {
+  // Force an update on route transition
+  // eslint-disable-next-line react/no-unused-prop-types
+  location: string;
+  children: React.ReactNode;
+};
+
+type State = {
+  routeAnnouncement: string;
+};
+
+class RouteAnnouncerWrapper extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      routeAnnouncement: '',
+    };
+  }
+
+  override componentDidUpdate(): void {
+    const {routeAnnouncement} = this.state;
+
+    requestAnimationFrame(() => {
+      const firstHeading = document.querySelectorAll(`#__docusaurus h1`)[0];
+      const pageName = firstHeading?.textContent ?? document.title;
+      const newAnnouncement = `Navigated to ${pageName}`;
+      const oldAnnouncement = routeAnnouncement;
+      if (oldAnnouncement !== newAnnouncement) {
+        this.setState({
+          routeAnnouncement: newAnnouncement,
+        });
+      }
+    });
+  }
+
+  override render(): JSX.Element {
+    const {children} = this.props;
+    const {routeAnnouncement} = this.state;
+    return (
+      <>
+        {children}
```

Comment:
does this component need a children? 

Can't it work standalone like `<RouteAnnouncer/>` ?

Afaik it doesn't provide any context to the children so that props seem unnecessary